frontend/tests/ui/util.py

```diff
@@ -297,3 +311,165 @@ def add_identities_group(driver, group_alias):
+
+    with driver.page_transition():
+        driver.wait.until(
+            EC.element_to_be_clickable(
+                (By.XPATH, ('//a[@href = "/course/new/"]')))).click()
+
+    driver.wait.until(
+        EC.visibility_of_element_located(
+            (By.CLASS_NAME, ('name')))).send_keys(course_alias)
+    driver.browser.find_element_by_class_name('alias').send_keys(
+        course_alias)
+    driver.typeahead_helper('*[contains(@class, "omegaup-course-details")]',
+                            school_name,
+                            select_suggestion=False)
+    driver.browser.find_element_by_tag_name('textarea').send_keys(
+        'course description')
+
+    if has_privileges:
+        with driver.page_transition():
+            driver.browser.find_element_by_tag_name('form').submit()
+        assert (('/course/%s/edit/' % course_alias) in
+                driver.browser.current_url), driver.browser.current_url
+    else:
+        submit_element = driver.browser.find_element_by_tag_name('form')
+        assert_get_alert(driver, submit_element)
+
+
+@contextlib.contextmanager
+def create_problem(driver, problem_alias, has_privileges=True):
+    '''Create a problem.'''
+
+    driver.wait.until(
+        EC.element_to_be_clickable(
+            (By.ID, 'nav-problems'))).click()
+    with driver.page_transition():
+        driver.wait.until(
+            EC.element_to_be_clickable(
+                (By.XPATH,
+                 ('//li[@id = "nav-problems"]'
+                  '//a[@href = "/problem/new/"]')))).click()
+
+    driver.wait.until(
+        EC.visibility_of_element_located(
+            (By.XPATH,
+             '//input[@name = "alias"]'))).send_keys(problem_alias)
+    driver.wait.until(
+        EC.visibility_of_element_located(
+            (By.XPATH,
+             '//input[@name = "title"]'))).send_keys(problem_alias)
+    input_limit = driver.wait.until(
+        EC.visibility_of_element_located(
+            (By.XPATH,
+             '//input[@name = "input_limit"]')))
+    input_limit.clear()
+    input_limit.send_keys('1024')
+    # Alias should be set automatically
+    driver.browser.find_element_by_name('source').send_keys('test')
+    # Make the problem public
+    driver.browser.find_element_by_xpath(
+        '//input[@name="visibility" and @value = "1"]').click()
+    contents_element = driver.browser.find_element_by_name(
+        'problem_contents')
+    contents_element.send_keys(os.path.join(
+        OMEGAUP_ROOT, 'frontend/tests/resources/triangulos.zip'))
+
+    if has_privileges:
+        with driver.page_transition(wait_for_ajax=False):
+            contents_element.submit()
+        assert (('/problem/%s/edit/' % problem_alias) in
+                driver.browser.current_url), driver.browser.current_url
+    else:
+        assert_get_alert(driver, contents_element)
+
+
+def assert_get_alert(driver, submit_element):
+    ''' When an identity tries to create something it should get a message. '''
```

Comment:
Esto no es necesariamente porque es una identidad. Simplemente que la identidad no tiene privilegios:

```suggestion
    '''When an identity tries to perform an action without permission, a message is shown.'''
```

Reply:
Listo